tox_pdm/plugin4.py

```diff
@@ -0,0 +1,232 @@
+"""Plugin specification for Tox 4"""
+from __future__ import annotations
+
+import os
+import shutil
+import sys
+import typing as t
+from pathlib import Path
+
+from tox.config.cli.parser import DEFAULT_VERBOSITY
+from tox.config.sets import EnvConfigSet
+from tox.execute.api import Execute
+from tox.execute.local_sub_process import LocalSubProcessExecuteInstance
+from tox.execute.request import StdinSource
+from tox.plugin import impl
+from tox.tox_env.package import Package, PackageToxEnv
+from tox.tox_env.python.api import Python, PythonInfo
+from tox.tox_env.python.package import WheelPackage
+from tox.tox_env.python.pip.pip_install import Pip
+from tox.tox_env.python.runner import PythonRun
+from virtualenv.discovery.builtin import get_interpreter
+
+from tox_pdm.utils import clone_pdm_files, get_env_lib_path, is_same_path, is_sub_array
+
+if t.TYPE_CHECKING:
+
+    from argparse import ArgumentParser
+
+    from tox.execute.api import ExecuteInstance, ExecuteOptions
+    from tox.execute.request import ExecuteRequest
+    from tox.execute.stream import SyncWrite
+    from tox.tox_env.api import ToxEnvCreateArgs
+    from tox.tox_env.register import ToxEnvRegister
+
+
+@impl
+def tox_add_option(parser: ArgumentParser) -> None:
+    os.environ["TOX_TESTENV_PASSENV"] = "PYTHONPATH"
```

Comment:
This is hacky and will likely mean users can no longer use this. How come you need to set this? Feels against the spirit of tox. If you must though extend this list https://github.com/tox-dev/tox/blob/rewrite/src/tox/tox_env/api.py#L176 (via subclass, call super and appending to it).